src/esy.ml

```diff
@@ -69,6 +71,7 @@ module Discover = struct
   let run ~dir : discover list Promise.t =
     let open Promise.Syntax in
     dir |> parse_dirs_up |> Promise.all_list >>| List.concat
+    >>| List.dedup_and_sort ~compare
```

Comment:
Shouldn't the de-duplication be on `parse_dir`? This is the function that is returning duplicates as far as I see.

Reply:
Yeah that makes more sense...